### ISSUE TITLE
Flink controller does not properly clean up all resources created when HA is enabled

### ISSUE THREAD

**Issue by martinbomio** [spotify/flink-on-k8s-operator#560]
Hello. We are creating clusters with high availability enabled. For some context, these clusters are created so that one and only one job will run on them.

We set this properties on the cluster:
```
    # High availability properties
    kubernetes.cluster-id: ....
    kubernetes.namespace: ....
    high-availability.cluster-id: ...
    high-availability: org.apache.flink.kubernetes.highavailability.KubernetesHaServicesFactory
    high-availability.storageDir: .....
```

The HA is enabled successfully and works without issues. We tried it by manually killing the JM and another will be brought up and the job will continue to be executed without issues.

But when we want to delete the cluster because we want to recreate it with a new property or new code, not all resources created by the HA (like the configmap) are being properly cleaned up, which means that when we relaunch the cluster, the old job will continue to exists, competing for resources with the new job being launched.

Is there a way for the controller to fully managed all resources created when HA is enabled?

**Comment by regadas** [comment]
Hi @martinbomio we actually talked about this today. We will be addressing this soon. Meanwhile, if this is really blocking you, you could potentially patch those configmaps with the right owner ref, but I understand this is a bit tricky;

**Comment by regadas** [comment]
Closed via #566
